Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/UserManager.java

```
@@ -297,29 +297,31 @@ public int activateUser(String sUserId, String sAdminId) throws QuadrigaStorageE
      * 
      * @param request
      *            The {@link AccountRequest} encapsulating the user's data.
-     * @return true if request was successfully added; otherwise false
+     * @return true if request was successfully added; false if username exists or the account needs to be approved by the admin.
      * @author jdamerow
      * @throws QuadrigaStorageException
-     * @throws UsernameExistsException
+     * @throws UsernameExistsException 
      * @throws QuadrigaNotificationException 
      */
     @Override
     @Transactional
-    public boolean addNewUser(AccountRequest request) throws QuadrigaStorageException, UsernameExistsException {
-        QuadrigaUserDTO userDTO = usermanagerDAO.getUserDTO(request.getUsername());
+    public boolean addNewUser(AccountRequest request) throws QuadrigaStorageException, UsernameExistsException{
 
         // Check if username is already in use
-        if (userDTO != null)
-            throw new UsernameExistsException("Username already in use.");
-
+        if (usermanagerDAO.getUserDTO(request.getUsername()) != null){
+            logger.debug("Username already in use.");
+            throw new UsernameExistsException();
+        }
         QuadrigaUserRequestsDTO userRequest = usermanagerDAO.getUserRequestDTO(request.getUsername());
-        if (userRequest != null)
-            throw new UsernameExistsException("Username already in use.");
-
-        String plainPassword = request.getPassword();
-        boolean success = usermanagerDAO.addNewUserAccountRequest(request.getUsername(), encryptPassword(plainPassword),
-                request.getName(), request.getEmail());
         
+        if (userRequest != null){
+            logger.debug("User account needs to be approved by the admin.");
+            return false;
```

this should throw an exception as it was before. The assumption here is that the username does not exist yet. If it does we have an unexpected situation which warrants an exception. Especially since you're not handling the case when it returns false in some cases (see comments below).